src/platform/plugins/shared/ai_assistant_management/selection/server/plugin.ts

```diff
@@ -38,27 +38,36 @@ export class AIAssistantManagementSelectionPlugin
     >
 {
   private readonly config: AIAssistantManagementSelectionConfig;
+  isServerless: boolean;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.config = initializerContext.config.get();
+    this.isServerless = initializerContext.env.packageInfo.buildFlavor === 'serverless';
   }
 
   public setup(
     core: CoreSetup,
     plugins: AIAssistantManagementSelectionPluginServerDependenciesSetup
   ) {
+    const envSpecificScope = this.isServerless ? 'Observability' : 'Observability and Search';
+
     core.uiSettings.register({
       [PREFERRED_AI_ASSISTANT_TYPE_SETTING_KEY]: {
-        name: i18n.translate('aiAssistantManagementSelection.preferredAIAssistantTypeSettingName', {
-          defaultMessage: 'AI Assistant for Observability and Search visibility',
-        }),
+        name: i18n.translate(
+          'aiAssistantManagementSelection.preferredAIAssistantTypeSettingName', // TODO: set different string for serverless or not
+          {
+            defaultMessage: 'AI Assistant for ' + envSpecificScope + ' visibility',
```

Comment:
The `defaultMessage` string must be static. In order to inject dynamic content you need to use template values: 
https://github.com/elastic/kibana/blob/d376e9bda24ba0470b1151f0285a4234ffa1eac0/x-pack/platform/packages/shared/kbn-ai-assistant/src/chat/chat_body.tsx#L614-L618